packages/components/src/components/PipelineRun/PipelineRun.js

```diff
@@ -242,7 +242,8 @@ export /* istanbul ignore next */ class PipelineRunContainer extends Component {
       );
     }
 
-    const pipelineRunName = pipelineRun.metadata.name;
+    const pipelineRunName =
+      pipelineRun.metadata.name || pipelineRun.metadata.generateName;
```

Comment:
Is there any chance that metadata could be undefined here?

Reply:
No it's always defined and contains one of either `name` or `generateName` at a minimum.